src/Marten.AspNetCore/QueryableExtensions.cs

```diff
@@ -276,4 +276,33 @@ public static class QueryableExtensions
         stream.Position = 0;
         await stream.CopyToAsync(context.Response.Body, context.RequestAborted).ConfigureAwait(false);
     }
+
+    /// <summary>
+    /// Write an raw SQL query result directly to the HttpContext
+    /// </summary>
+    /// <param name="session"></param>
+    /// <param name="sql"></param>
+    /// <param name="context"></param>
+    /// <param name="contentType"></param>
+    /// <param name="onFoundStatus">Defaults to 200</param>
+    /// <returns></returns>
+    public static async Task WriteJson(
+        this IQuerySession session,
+        string sql,
+        HttpContext context,
+        string contentType = "application/json",
+        int onFoundStatus = 200,
+        params object[] parameters
+    )
+    {
+        var stream = new MemoryStream();
```

Comment:
Not related to your PR:

We should find a way to avoid the temporary use of MemoryStream to avoid the allocations and the copying.

I once prototyped something but (unfortunately) didn't continue on that.
Is there interested to follow along with this?
If so I'd create a issue (and later when I'm not so busy I'll can look into that).

Reply:
@gfoidl that's fine by me; if you're open to contribution, then such performance optimisation would be more than welcome.

Reply:
Sure.

Sometimes I wonder by myself what a query in a repo unveils: https://github.com/JasperFx/marten/discussions/2408
Which I had already forgotten...

Can that discussion be converted to an issue and assigned to me (so that I won't forget it anymore, as AFAIK discussions can't be assigned, etc.).
But note: at the moment I'm very busy, so I'll be able to work on that beginning next year (or maybe over the X-mas holidays...)

Reply:
@gfoidl I cannot convert discussion into an issue, but I created issue based on discussion https://github.com/JasperFx/marten/issues/2791 :)